x-pack/plugins/enterprise_search/server/routes/enterprise_search/connectors.ts

```diff
@@ -696,4 +698,49 @@ export function registerConnectorRoutes({ router, log }: RouteDependencies) {
       return response.ok();
     })
   );
+
+  router.get(
+    {
+      path: '/internal/enterprise_search/connectors/available_indices',
+      validate: {
+        query: schema.object({
+          from: schema.number({ defaultValue: 0, min: 0 }),
+          search_query: schema.maybe(schema.string()),
+          size: schema.number({ defaultValue: 40, min: 0 }),
+        }),
+      },
+    },
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const { from, size, search_query: searchQuery } = request.query;
+      const { client } = (await context.core).elasticsearch;
+
+      const { indexNames, indices, totalResults } = await fetchConnectorIndices(
+        client,
+        searchQuery,
+        from,
+        size
+      );
+      const connectors = await fetchConnectors(client.asCurrentUser, indexNames);
+      const crawlers = await fetchCrawlers(client, indexNames);
+      const enrichedIndices = indices.map((index) => ({
```

Comment:
wait why are we returning connectors and crawler data here? Aren't we only returning indices that don't have a crawler or connector attached?

Reply:
it is a copy, thats why. These should be trimmed.